File: gto-support-recyclerview/src/main/java/org/ccci/gto/android/common/recyclerview/adapter/DataBindingViewHolder.java
package org.ccci.gto.android.common.recyclerview.adapter;

import androidx.annotation.NonNull;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

public class DataBindingViewHolder<B extends ViewDataBinding> extends RecyclerView.ViewHolder {
    @NonNull
    public final B binding;

    public DataBindingViewHolder(@NonNull final B binding) {
        super(binding.getRoot());
        this.binding = binding;
    }
}

File: gto-support-recyclerview/src/main/kotlin/org/ccci/gto/android/common/recyclerview/adapter/DataBindingViewHolder.kt
package org.ccci.gto.android.common.recyclerview.adapter

import androidx.databinding.ViewDataBinding
import org.ccci.gto.android.common.androidx.recyclerview.adapter.DataBindingViewHolder

@Deprecated("Since v3.11.2, use DataBindingViewHolder from gto-support-androidx-recyclerview instead.")
open class DataBindingViewHolder<B : ViewDataBinding>(binding: B) : DataBindingViewHolder<B>(binding)
